napari/utils/_magicgui.py

```diff
@@ -53,6 +54,10 @@ def register_types_with_magicgui():
             Viewer. And expects the user to return a list of layer data tuples.
 
     """
+    from magicgui.widgets import FunctionGui
+
+    Source.update_forward_refs(FunctionGui=FunctionGui)
```

Comment:
What's the story here?

Reply:
will add a comment to the code, but here's the gist:
In `_source.py` I defined the `widget` field using a forward reference so as to avoid unnecessarily importing magicgui when we define `Source`.  The line here will "resolve" that forward reference to an actual type, and I did it in `_magicgui.py` since that's where we actually know we will have imported magicgui.

relevant pydantic docs: https://pydantic-docs.helpmanual.io/usage/postponed_annotations/

Reply:
added